example/gluon/embedding_learning/model.py

```diff
@@ -110,7 +110,10 @@ def hybrid_forward(self, F, x):
             mask[i:i+k, i:i+k] = 0
 
         weights = weights * F.array(mask) * (distance < self.nonzero_loss_cutoff)
-        weights = weights / F.sum(weights, axis=1, keepdims=True)
+        weight_norm = F.sum(weights, axis=1, keepdims=True)
+        # we need to change all zero elements to 1 to avoid division by zero
+        weight_norm = F.where(weight_norm == 0, F.ones_like(weight_norm), weight_norm)
```

Comment:
Feels like a hack. ping @chaoyuaw

Reply:
I see some more issues in the same file, let me come up with a complete fix.